services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -131,6 +170,11 @@ async def list_files(
             )
 
             for fmd in file_metadatas:
+                if fmd.is_directory and expand_dirs:
+                    # lists content of the directory
+                    data.extend(await self._expand_directory(fmd))
```

Comment:
here I'm a bit scarred. it looks like you're creating a **huge** list of file meta data. How is the memory going with this?
If I have many folders, they will all be filled with up to 1000 files right? I think that does not go well with storage not crashing... and remember there is only **one** storage running. If this guy crashes, all the dy-sidecars that might be uploading will fail, all the comp. pipeline running at that time point will probably fail, and all the frontend upload will fail... let me know if I'm wrong here

Reply:
this function needs to be capped